sdk/go/common/resource/plugin/context.go

```diff
@@ -113,14 +115,23 @@ func (ctx *Context) Request() context.Context {
 	}
 	c = opentracing.ContextWithSpan(c, ctx.tracingSpan)
 	c, cancel := context.WithCancel(c)
+	ctx.cancelLock.Lock()
 	ctx.cancelFuncs = append(ctx.cancelFuncs, cancel)
+	ctx.cancelLock.Unlock()
 	return c
 }
 
 // Close reclaims all resources associated with this context.
 func (ctx *Context) Close() error {
 	defer func() {
-		for _, cancel := range ctx.cancelFuncs {
+		// It is possible that cancelFuncs may be appended to while this function is running.
```

Comment:
typo:
```suggestion
		// It is possible that cancelFuncs may be appended while this function is running.
```

Reply:
Will fix. It's a midwesternism that I forget isn't universal.

I.e. "Would you like to come with?"